content/sensu-go/5.17/api/secrets.md

```diff
@@ -0,0 +1,386 @@
+-H "Authorization: Bearer $SENSU_ACCESS_TOKEN"
+
+HTTP/1.1 200 OK
+[
+  {
+    "type": "Secret",
+    "api_version": "secrets/v1",
+    "metadata": {
+      "name": "sensu-ansible-token",
+      "namespace": "default"
+    },
+    "spec": {
+      "id": "ANSIBLE_TOKEN",
+      "provider": "env"
+    }
+  }
+]
+{{< /highlight >}}
+
+#### API Specification {#secrets-get-specification}
+
+/secrets (GET)  | 
+---------------|------
+description    | Returns the list of secrets for the specified namespace. `id` is the identifying key for the provider to retrieve the secret. `provider` is the name of the provider with the secret.
+example url    | http://hostname:8080/api/enterprise/secrets/v1/namespaces/default/secrets
+response type  | Array
+response codes | <ul><li>**Success**: 200 (OK)</li><li>**Error**: 500 (Internal Server Error)</li></ul>
+output         | {{< highlight shell >}}
+[
+  {
+    "type": "Secret",
+    "api_version": "secrets/v1",
+    "metadata": {
+      "name": "sensu-ansible-token",
+      "namespace": "default"
+    },
+    "spec": {
+      "id": "ANSIBLE_TOKEN",
+      "provider": "env"
+    }
+  }
+]
+{{< /highlight >}}
+
+## The `/secrets/:secret` endpoint {#the-secrets-secret-endpoint}
+
+### `/secrets/:secret` (GET) {#secrets-secret-get}
+
+The `/secrets/:secret` API endpoint provides HTTP GET access to data for a specific `secret`, by secret name.
+
+#### EXAMPLE {#secrets-secret-get-example}
+
+In the following example, querying the `/secrets/:secret` API endpoint returns a JSON map that contains the requested `:secret`.
+
+{{< highlight shell >}}
+curl -X GET \
+http://127.0.0.1:8080/api/enterprise/secrets/v1/namespaces/default/secrets/sensu-ansible-token \
+-H "Authorization: Bearer $SENSU_ACCESS_TOKEN"
+
+HTTP/1.1 200 OK
+[
+  {
+    "type": "Secret",
+    "api_version": "secrets/v1",
+    "metadata": {
+      "name": "sensu-ansible-token",
+      "namespace": "default"
+    },
+    "spec": {
+      "id": "ANSIBLE_TOKEN",
+      "provider": "env"
+    }
+  }
+]
+{{< /highlight >}}
+
+#### API Specification {#secrets-secret-get-specification}
+
+/secrets/:secret (GET) | 
+---------------------|------
+description          | Returns the specified secret. `id` is the identifying key for the provider to retrieve the secret. `provider` is the name of the provider with the secret.
+example url          | http://hostname:8080/api/enterprise/secrets/v1/namespaces/default/secrets/sensu-ansible-token
+response type        | Map
+response codes       | <ul><li>**Success**: 200 (OK)</li><li> **Missing**: 404 (Not Found)</li><li>**Error**: 500 (Internal Server Error)</li></ul>
+output               | {{< highlight json >}}
+[
+  {
+    "type": "Secret",
+    "api_version": "secrets/v1",
+    "metadata": {
+      "name": "sensu-ansible-token",
+      "namespace": "default"
+    },
+    "spec": {
+      "id": "ANSIBLE_TOKEN",
+      "provider": "env"
+    }
+  }
+]
```

Comment:
When getting a single secret `/secrets/:secret` it should be a single JSON object, not a JSON list of objects.
```
{
    "type": "Secret",
    "api_version": "secrets/v1",
    "metadata": {
        "name": "sensu-ansible-token",
        "namespace": "default"
    },
    "spec": {
        "id": "ANSIBLE_TOKEN",
        "provider": "env"
    }
}
```